numba/np/arrayobj.py

```diff
@@ -7035,6 +7035,44 @@ def impl(a, axis1, axis2):
     return impl
 
 
+@overload(np.moveaxis)
+def numpy_moveaxis(a, source, destination):
+    if not isinstance(a, types.Array):
+        raise errors.TypingError('The first argument "a" must be an array')
+    if not isinstance(source, (types.Integer, types.BaseTuple)):
+        raise errors.TypingError(
+            'The second argument "source" must be an integer or tuple of integers'
+        )
+    if not isinstance(destination, (types.Integer, types.BaseTuple)):
+        raise errors.TypingError(
+            'The third argument "destination" must be an integer or tuple of integers'
+        )
+
+    ndim = a.ndim
+    order_init = (1,) * a.ndim
+
+    def impl(a, source, destination):
+        source = normalize_axis_tuple("np.moveaxis", "source", ndim, source)
+        destination = normalize_axis_tuple("np.moveaxis", "destination", ndim, destination)
+        if len(source) != len(destination):
+            raise ValueError(
+                "`source` and `destination` arguments must have the same number of parameters"
+            )
+
+        order_list = [n for n in range(a.ndim) if n not in source]
+
+        for dest, src in sorted(zip(destination, source)):
+            order_list.insert(dest, src)
```

Comment:
I assume there is no tuple_insert helper? If there is (or it is easy to add) we can implement in a single pass